lib/get-selenium-status-url.js

```diff
@@ -1,36 +1,71 @@
-module.exports = getSeleniumStatusUrl;
-
 var URI = require('urijs');
+var PROCESS_TYPES = exports.PROCESS_TYPES = {
+  STANDALONE: 0,
+  GRID_HUB: 1,
+  GRID_NODE: 2
+};
 
-function getSeleniumStatusUrl(seleniumArgs) {
-  var port = 4444;
-  var hostname = 'localhost';
-  var portArg = seleniumArgs.indexOf('-port');
-  var hubArg = seleniumArgs.indexOf('-hub');
+exports.getRunningProcessType = function(seleniumArgs) {
   var roleArg = seleniumArgs.indexOf('-role');
+  var role = (roleArg !== -1) ? seleniumArgs[roleArg + 1] : undefined;
 
-  if (hubArg !== -1) {
-    hubURI = new URI(seleniumArgs[hubArg + 1]);
-    hostname = hubURI.hostname() || hostname;
-    port = hubURI.port() || port;
-  }
+  if (roleArg === -1) return PROCESS_TYPES.STANDALONE;
+  else if (role === 'hub') return PROCESS_TYPES.GRID_HUB;
+  else if (role === 'node') return PROCESS_TYPES.GRID_NODE;
+  else return undefined;
+}
+
+exports.getSeleniumStatusUrl = function(seleniumArgs) {
+  var processType = this.getRunningProcessType(seleniumArgs);
+  var portArg = seleniumArgs.indexOf('-port');
+  var port = (portArg !== -1) ? seleniumArgs[portArg + 1] : undefined;
 
-  if (roleArg !== -1) {
-    switch(seleniumArgs[roleArg + 1]){
-      case 'node':
-        if(hubArg === -1){
-          port = portArg !== -1 ? seleniumArgs[portArg + 1] : 5555;
-        }
-        return 'http://' + hostname + ':' + port + '/wd/hub/status';
-      case 'hub':
-        port = portArg !== -1 ? seleniumArgs[portArg + 1] : 4444;
-        return 'http://localhost:' + port + '/api/grid/hub/';
-    }
+  var statusURI = new URI('http://localhost');
+  var nodeStatusAPIPath = '/wd/hub/status';
+  var hubStatusAPIPath = '/grid/api/hub';
+
+  switch (processType) {
+    case PROCESS_TYPES.STANDALONE:
+      statusURI.port(4444);
+      statusURI.path(nodeStatusAPIPath);
+      break;
+    case PROCESS_TYPES.GRID_HUB:
+      statusURI.port(4444);
+      statusURI.path(hubStatusAPIPath);
+      break;
+    case PROCESS_TYPES.GRID_NODE:
+      statusURI.port(5555);
+      statusURI.path(nodeStatusAPIPath);
+      break;
+    default:
+      throw 'ERROR: Trying to run selenium in an unknown way.';
   }
 
+  // Running with a non-default port
   if (portArg !== -1) {
-    port = seleniumArgs[portArg + 1];
+    statusURI.port(seleniumArgs[portArg + 1]);
   }
 
-  return 'http://' + hostname + ':' + port + '/wd/hub/status';
+  return statusURI.toString();
+}
+
+exports.getRemoteSeleniumHubStatusUrl = function(seleniumArgs) {
```

Comment:
Originally my thinking was that on startup you might want to check that you are able to connect to the local process, and also check to see if you could connect to the remote hub (if there is one). If either of those connections cannot be made, then startup would be considered a failure. Since that is a change in behavior from the old code, I didn't added the hub check and only left the unused function.

There is probably a better second level check to be making instead of "is the remote hub responding". Something like "is the remote hub responding and is this node connected to said hub". I don't immediately see that type of API in the docs I found, so I'm going to back out the unused code instead (a244c49).